build-tools/packages/build-cli/src/commands/modify/fluid-imports.ts

```diff
@@ -41,25 +44,273 @@ export default class UpdateFluidImportsCommand extends BaseCommand<
+					// Any other prospects should be inserted after this now.
+					for (const otherProspects of takeOverProspects.slice(1)) {
+						otherProspects.insertAfterIndex = true;
+					}
+					// Remove the missing as it is now in place.
+					this.missingImports.splice(this.missingImports.indexOf(replacement), 1);
+					// We could remove the existing entry now, but that would
+					// alter the array being iterated. No further meaningful use is expected.
+					// The later removal check will skip as it now has imports.
+					// Set originallyUnassigned as an ounce of precaution.
+					fluidImport.originallyUnassigned = true;
+				}
+			}
+		}
+
+		const reverseSortedAdditions = this.missingImports.sort((a, b) => {
+			const indexDelta = b.index - a.index;
+			if (indexDelta) return indexDelta;
+			return b.order - a.order;
+		});
+		for (const addition of reverseSortedAdditions) {
+			// Note that ts-morph will not preserve blank lines that may have existed
+			// near the insertion point.
+			// When inserting before it is likely desirable to capture any for the
+			// leading trivia (comments included) and "move" them to the inserted
+			// import. Likewise for inserting after the leading trivia of the next
+			// import should be moved to the trailing trivia of the inserted.
+			// ts-morph has some fairly unexpected results with manipulations so
+			// some trial and error may be required to get desired behavior.
+			this.log.verbose(
+				`\tInjecting ${addition.declaration.moduleSpecifier} ${
+					addition.insertAfterIndex ? "after" : "before"
+				} ${this.sourceFile
+					.getImportDeclarations()
+					[addition.index].getModuleSpecifierValue()}`,
+			);
+			this.sourceFile.insertImportDeclaration(
+				addition.index + (addition.insertAfterIndex ? 1 : 0),
+				addition.declaration,
+			);
+		}
+
+		// Check for import that has no imports, default or named, that has been
+		// modified. And only after insertions have been taken care of.
+		for (const { importDeclaration, originallyUnassigned } of this.fluidImports) {
+			if (!originallyUnassigned && isImportUnassigned(importDeclaration)) {
+				importDeclaration.remove();
+			}
+		}
+
+		return true;
+	}
+
+	private ensureFluidImport({
+		packageName,
+		level,
+		isTypeOnly,
+	}: {
+		packageName: string;
+		level: ApiLevel;
+		isTypeOnly: boolean;
+	}): FluidImportData {
+		const match = (element: FluidImportData): boolean =>
+			element.packageName === packageName &&
+			element.level === level &&
+			element.declaration.isTypeOnly === isTypeOnly;
+
+		const preexisting = this.fluidImports.find(match);
+		if (preexisting !== undefined) {
+			return preexisting;
+		}
+
+		// Check for a pending import
+		const existing = this.missingImports.find(match);
+		if (existing !== undefined) {
+			return existing;
+		}
+
+		const moduleSpecifier = level === publicLevel ? packageName : `${packageName}/${level}`;
+		const order = knownLevels.indexOf(level) * 2 + (isTypeOnly ? 0 : 1);
```

Comment:
I've tried to figure out why there's a magic x2 in there unsuccessfully. Worth a comment.